create-aleo-app/template-extension/src/service_worker.js

```diff
@@ -0,0 +1,20 @@
+async function createOffscreen(path) {
+    const offscreenUrl = chrome.runtime.getURL(path);
+
+    const existingContexts = await chrome.runtime.getContexts({
+        contextTypes: ["OFFSCREEN_DOCUMENT"],
+        documentUrls: [offscreenUrl]
+    });
+
+    if (existingContexts.length > 0) {
+        return;
+    }
+
+    await chrome.offscreen.createDocument({
+        url: offscreenUrl,
+        reasons: ["WORKERS"],
+        justification: "Top-level await and Workers cannot be used in service workers, the V3 extension system is horrendously awful.",
```

Comment:
lol but maybe something a little more neutral?

```suggestion
        justification: "Enable usage of the Aleo WASM multi-threading via Workers.",
```